controllers/datafoundation_client.go

```diff
@@ -0,0 +1,241 @@
+	*ManagedFusionOfferingReconciler
+
+	offering                      *v1alpha1.ManagedFusionOffering
+	dataFoundationClientSpec      dataFoundationClientSpec
+	storageClient                 ocsclient.StorageClient
+	defaultBlockStorageClassClaim ocsclient.StorageClassClaim
+	defaultFSStorageClassClaim    ocsclient.StorageClassClaim
+}
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclients,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=operators.coreos.com,namespace=system,resources=clusterserviceversions,verbs=get;list;watch;delete;update;patch
+
+func dfcSetupWatches(controllerBuilder *builder.Builder) {
+	csvPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(res client.Object) bool {
+				return strings.HasPrefix(res.GetName(), ocsClientOperatorName)
+			},
+		),
+	)
+	controllerBuilder.
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
+		Owns(&ocsclient.StorageClient{}).
+		Owns(&ocsclient.StorageClassClaim{})
+}
+
+func DFCAddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(ocsclient.AddToScheme(scheme))
+}
+
+func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	r := dataFoundationClientReconciler{}
+	r.initReconciler(offeringReconciler, offering)
+
+	if err := r.parseSpec(offering); err != nil {
+		return ctrl.Result{}, err
+	}
+	return r.reconcilePhases()
+}
+
+func (r *dataFoundationClientReconciler) initReconciler(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
+	r.ManagedFusionOfferingReconciler = reconciler
+	r.offering = offering
+
+	r.storageClient.Name = "ocs-storageclient"
+	r.storageClient.Namespace = offering.Namespace
+
+	r.defaultBlockStorageClassClaim.Name = "default-ocs-block-storageclass-claim"
+	r.defaultBlockStorageClassClaim.Namespace = offering.Namespace
+
+	r.defaultFSStorageClassClaim.Name = "default-ocs-fs-storageclass-claim"
+	r.defaultFSStorageClassClaim.Namespace = offering.Namespace
+}
+
+func (r *dataFoundationClientReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
+	r.Log.Info("Parsing ManagedFusionOffering Data Foundation Client spec")
+
+	isValid := true
+	onboardingTicket, found := offering.Spec.Config["onboardingTicket"]
+	if !found {
+		r.Log.Error(
+			fmt.Errorf("missing field: onboardingTicket"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	} else if onboardingTicket == "" {
+		r.Log.Error(
+			fmt.Errorf("onboardingTicket is empty"),
+			"an error occurred while parsing ManagedFusionOffering Data Foundation Client spec",
+		)
+		isValid = false
+	}
+
+	dfStorageProviderEndpoint, found := offering.Spec.Config["dataFoundationProviderEndpoint"]
```

Comment:
See one of the previous comments for explanation.  
```suggestion
	providerEndpoint, found := offering.Spec.Config["dataFoundationProviderEndpoint"]
```

Reply:
updated